packages/arb-util/fireblocks/fireblocks.go

```diff
@@ -0,0 +1,512 @@
+}
+
+func (fb *Fireblocks) sendRequest(method string, path string, params url.Values, body interface{}) (*http.Response, error) {
+	var jsonData []byte
+	if body != nil {
+		var err error
+		jsonData, err = json.Marshal(body)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	logger.Debug().RawJSON("request", jsonData).Msg("creating new fireblocks transaction")
+	for i := 0; i < 3; i++ {
+		resp, err := fb.sendRequestImpl(method, path, params, jsonData)
+		if err != nil && strings.Contains(err.Error(), "nonce was already used") {
+			// Duplicate nonce used, try again
+			logger.Warn().Msg("duplicate nonce sent to fireblocks")
+			continue
+		}
+
+		return resp, err
+	}
+
+	return nil, errors.New("too many fireblocks duplicate nonce errors")
+}
+func (fb *Fireblocks) sendRequestImpl(method string, path string, params url.Values, body []byte) (*http.Response, error) {
+	token, err := fb.signJWT(path, body)
+	if err != nil {
+		return nil, err
+	}
+
+	uri, err := url.ParseRequestURI(fb.baseUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	uri.Path = path
+	uri.RawQuery = params.Encode()
+
+	client := &http.Client{}
+	var req *http.Request
+	req, err = http.NewRequest(method, uri.String(), bytes.NewBuffer(body))
+	if err != nil {
+		logger.
+			Error().
+			Err(err).
+			Str("url", uri.String()).
+			Str("method", method).
+			Msg("error creating new fireblocks request")
+		return nil, err
+	}
+	if method == http.MethodPost {
+		req.Header.Set("Content-Type", "application/json")
+	}
+	req.Header.Set("Accept", "*/*")
+	req.Header.Set("X-API-Key", fb.apiKey)
+	req.Header.Set("Authorization", "Bearer "+token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.
+			Error().
+			Err(err).
+			Str("url", fb.baseUrl).
+			Msg("error doing fireblocks request")
+		return nil, err
+	}
+
+	if resp.StatusCode >= 300 {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			logger.
+				Error().
+				Err(err).
+				Str("url", uri.String()).
+				Str("status", resp.Status).
+				Msg("error reading body after sending fireblocks request")
+			return nil, fmt.Errorf("status '%s' receiving fireblocks response", resp.Status)
+		}
+
+		bodyStr := string(body)
+		if strings.Contains(bodyStr, "nonce was already used") {
+			logger.
+				Warn().
+				Str("url", uri.String()).
+				Str("status", resp.Status).
+				Str("body", bodyStr).
+				Msg("error returned when posting fireblocks request")
+			return nil, fmt.Errorf("nonce was already used")
+		}
+
+		logger.
+			Error().
+			Str("url", uri.String()).
+			Str("status", resp.Status).
+			Str("body", bodyStr).
+			Msg("error returned when posting fireblocks request")
+		return nil, fmt.Errorf("status '%s' fireblocks response", resp.Status)
+	}
+
+	return resp, nil
+}
+
+func (fb *Fireblocks) signJWT(path string, body []byte) (string, error) {
+	newPath := strings.Replace(path, "[", "%5B", -1)
+	newPath = strings.Replace(newPath, "]", "%5D", -1)
```

Comment:
Go seems to have a URL escape in the standard library if that's what you're looking for: https://pkg.go.dev/net/url#PathEscape

Reply:
newPath is never used in an actual URL, it is just used in generating signature.  I wanted to duplicate the exact same thing that the other fireblocks API libraries do to ensure I came up with the exact same signature.